qa/workunits/rgw/test_rgw_reshard.py

```diff
@@ -313,13 +315,22 @@ def boto_connect(portnum, ssl, proto):
     json_op = json.loads(cmd.decode('utf-8', 'ignore')) # ignore utf-8 can't decode 0x80
     assert len(json_op) == 0
 
+    # TESTCASE 'check that PUT succeeds during reshard'
+    log.debug(' test: PUT succeeds during reshard')
+    num_shards = get_bucket_stats(VER_BUCKET_NAME).num_shards
+    exec_cmd('''radosgw-admin --rgw-debug-reshard-completion-delay 5 \
+                      bucket reshard --bucket {} --num-shards {}'''
```

Comment:
these reshard tests already make heavy use of a `FaultInjector` class to test error paths and crash recovery. in https://github.com/ceph/ceph/pull/51733, i just added the ability to inject delays as well

that would allow this test case to use `--inject-delay-at=do_reshard --inject-delay-ms=5000` without any changes to the reshard code itself

Reply:
@cbodley that makes sense and is a neat concept. I'm a bit confused by the statement about this being usable "without any changes to the reshard code itself", though? Is there some dynamic nature to this fault injector code that I'm missing or do I still need to add a fault injector hook in `do_reshard` to trigger this delay at the appropriate place?

Reply:
injecting a sleep at the existing 'do_reshard' check (https://github.com/ceph/ceph/blob/3bc1ea5/src/rgw/driver/rados/rgw_reshard.cc#L972-L974) should behave the same as the sleep you added inside of `do_reshard()`. the bucket index is locked for resharding over all of this, so the racing `put_object()` call would be blocked either way

Reply:
Ah, I'd missed that there was already an existing 'do_reshard' check. I've included your commits and updated this to  use the FaultInjector now. It seems to be working as advertised.